### PR TITLE
Component/5204-Avatars

### PR DESCRIPTION
**Pull request by brianacnguyen**
**Description**

- Added AvatarBlockies, AvatarImage, AvatarInitial, and AvatarJazzIcon
- Replaced AvatarAccount with Avatar, AvatarNetwork with Network, AvatarToken with Token, AvatarIcon with IconInACircle, and AvatarFavicon with Favicon
- Added CirclePattern
- Updated Badges
- Added Badges to CirclePattern and all inherited components
- Updated stories, tests, and readmes for all affected components

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

### CODE REVIEW

File: app/component-library/components/Pickers/PickerAccount/PickerAccount.tsx [hunk 1]
```diff
@@ -5,35 +5,45 @@ import React from 'react';
...
-      />
+      {accountAvatarType === AvatarAccountType.Blockies && (
+        <Avatar
+          variant={AvatarVariants.Image}
+          imageSource={{ uri: toDataUrl(accountAddress) }}
+          size={AvatarSizes.Md}
+          style={styles.accountAvatar}
+        />
+      )}
+      {accountAvatarType === AvatarAccountType.JazzIcon && (
```
Comment (Cal-L) [inline app/component-library/components/Pickers/PickerAccount/PickerAccount.tsx:39]: Need to fix this condition and import Jazzicon avatar from app level

File: app/component-library/components/Pickers/PickerNetwork/PickerNetwork.tsx [hunk 2]
```diff
@@ -26,9 +26,8 @@ const PickerNetwork = ({
   return (
     <TouchableOpacity style={styles.base} onPress={onPress} {...props}>
       <Avatar
-        variant={AvatarVariants.Network}
-        size={AvatarSize.Xs}
-        name={label}
+        variant={AvatarVariants.Image}
```
Comment (Cal-L) [inline app/component-library/components/Pickers/PickerNetwork/PickerNetwork.tsx:29]: Eventually needs to use AvatarNetwork since we need label support

Comment (Cal-L) [reply app/component-library/components/Pickers/PickerNetwork/PickerNetwork.tsx:29]: Can you add TODO comments here and other places where this needs to happen

File: app/component-library/components/Tags/TagUrl/TagUrl.tsx [hunk 3]
```diff
@@ -19,9 +19,9 @@ const TagUrl = ({ imageSource, label, cta, style, ...props }: TagUrlProps) => {
   return (
     <View style={styles.base} {...props}>
       <Avatar
-        variant={AvatarVariants.Favicon}
+        variant={AvatarVariants.Image}
```
Comment (Cal-L) [inline app/component-library/components/Tags/TagUrl/TagUrl.tsx:22]: Need to use AvatarFavicon for now or add fallback similar to how avatar favicon works. (globe icon)

File: app/component-library/components-temp/Accounts/AccountBase/AccountBase.tsx [hunk 4]
```diff
@@ -23,9 +24,9 @@ const AccountBase = ({
     <View style={styles.container}>
       <BadgeWrapper badgeProps={badgeProps} style={styles.badgeWrapper}>
         <Avatar
-          variant={AvatarVariants.Account}
+          variant={AvatarVariants.Image}
```
Comment (Cal-L) [inline app/component-library/components-temp/Accounts/AccountBase/AccountBase.tsx:27]: Need to use AvatarAccount for now and support blockies + jazzicon

File: app/components/UI/Avatars/Avatar/Avatar.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+// Internal dependencies.
+import { AvatarProps, AvatarVariants } from './Avatar.types';
+
+const Avatar = (avatarProps: AvatarProps) => {
+  switch (avatarProps.variant) {
+    case AvatarVariants.Account:
+      return <AvatarAccount {...avatarProps} />;
+    case AvatarVariants.Favicon:
+      return <AvatarFavicon {...avatarProps} />;
+    case AvatarVariants.JazzIcon:
```
Comment (Cal-L) [inline app/components/UI/Avatars/Avatar/Avatar.tsx:20]: Lets remove exposing Jazzicon on this level and let AvatarAccount handle it

File: app/components/UI/Avatars/Avatar/Avatar.types.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import { AvatarNetworkProps } from './variants/AvatarNetwork/AvatarNetwork.types';
+import { AvatarTokenProps } from './variants/AvatarToken/AvatarToken.types';
+
+/**
+ * Avatar variants.
+ */
+export enum AvatarVariants {
+  Account = 'Account',
+  Favicon = 'Favicon',
+  JazzIcon = 'JazzIcon',
```
Comment (Cal-L) [inline app/components/UI/Avatars/Avatar/Avatar.types.ts:14]: Remove Jazzicon from here

File: app/components/UI/Avatars/Avatar/README.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+  badgeProps={badgeProps}
+  badgePosition={AvatarBadgePositions.BottomRight}
+  type={AvatarAccountType.Jazzicon}
+  accountAddress={SAMPLE_ACCOUNT_ADDRESS}/>;
+
+<Avatar
+  variant={AvatarVariants.Favicon}
+  size={AvatarSizes.Md}
+  backgroundColor={'#000000'}
+  isBadgeIncluded
```
Comment (Cal-L) [inline app/components/UI/Avatars/Avatar/README.md:165]: Can remove redundant boolean

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: app/component-library/components/Toast/Toast.tsx [hunk 8]
```diff
@@ -128,10 +127,9 @@ const Toast = forwardRef((_, ref: React.ForwardedRef<ToastRef>) => {
         const { accountAddress } = toastOptions;
         return (
           <Avatar
-            variant={AvatarVariants.Account}
-            accountAddress={accountAddress}
-            type={AvatarAccountType.JazzIcon}
-            size={AvatarSize.Md}
+            variant={AvatarVariants.JazzIcon}
```
Comment (Cal-L) [inline app/component-library/components/Toast/Toast.tsx:130]: @jpcloureiro Can you share your changes that introduces avatar type to the toast component?

File: app/component-library/components/Tags/TagUrl/TagUrl.tsx (hunk 3 above)
Comment (Cal-L) [inline app/component-library/components/Tags/TagUrl/TagUrl.tsx:22]: Replace with `Favicon`

**Review by Cal-L — COMMENTED** [review]
Left some more comments. Starting to look super clean!

**Review by georgewrmarshall — COMMENTED** [review]
This is a big PR. Would it not have been slightly more digestible to break it out into sections?
- The Circle pattern in one PR.
- Adding Network and Token components in another PR (whilst adding a deprecation notice to AvatarToken and AvatarNetwork)
- Replacing instances of AvatarToken and AvatarNetwork with Network and Token in one PR
- Update the Avatar components in another PR

Otherwise it's making sense.

File: app/component-library/components/Toast/Toast.tsx (hunk 8 above)
Comment (jpcloureiro) [reply app/component-library/components/Toast/Toast.tsx:131]: In order to allow the Toast to respect the global setting for the Account Avatar (JazzIcon or Blockies), we need to accept it as a Toast Prop, e.g. ` const { accountAvatarVariant } = toastOptions; ` and pass it in as the variant Avatar prop

File: app/component-library/components/Icons/IconContainer/IconContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+import { IconContainerProps } from './IconContainer.types';
+import {
+  DEFAULT_ICON_CONTAINER_SIZE,
+  ICON_SIZE_BY_ICON_CONTAINER_SIZE,
+  ICON_CONTAINER_TEST_ID,
+  ICON_CONTAINER_ICON_TEST_ID,
+} from './IconContainer.constants';
+import styleSheet from './IconContainer.styles';
+
+const IconContainer = ({
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply app/component-library/components/Icons/IconContainer/IconContainer.tsx:19]: I think IconContainer is appropriate. IMO an icon should remain a simple svg without a containing node

**Comment by brianacnguyen** [comment]
> 

I can split it out if need be. NBD to me

**Comment by brianacnguyen** [comment]
Will abandon this PR to split it out

**Pull request closed without merging**